android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java

```diff
@@ -217,6 +224,125 @@ public void initializeTelecomManager() {
         telecomManager = (TelecomManager) context.getSystemService(Context.TELECOM_SERVICE);
     }
 
+
+
+     /**
+       * Monitors and logs phone call activities, and shows the phone state
+       */
+     private class LegacyCallStateListener extends PhoneStateListener {
+
+         @Override
+         public void onCallStateChanged(int state, String incomingNumber) {
+             switch (state) {
+                 case TelephonyManager.CALL_STATE_RINGING:
+                     // Incoming call is ringing (not used for outgoing call).
+                     Log.i("onCallStateChanged", "CALL_STATE_RINGING");
+                     break;
+                 case TelephonyManager.CALL_STATE_OFFHOOK:
+                     // Phone call is active -- off the hook.
+                      // Check if there is active call in native
+                      boolean isInManagedCall = RNCallKeepModule.this.checkIsInManagedCall();
+
+                      // Only let the JS side know if there is active app call & active native call
+                      if(RNCallKeepModule.this.hasActiveCall && isInManagedCall){
+                          WritableMap args = Arguments.createMap();
+                          RNCallKeepModule.this.sendEventToJS("RNCallKeepHasActiveCall",args);
+                      }else if(VoiceConnectionService.currentConnections.size() > 0){
+                        // Will enter here for the first time to mark the app has active call
+                          RNCallKeepModule.this.hasActiveCall = true;
+                      }
+                     Log.i("onCallStateChanged", "CALL_STATE_OFFHOOK");
+                     break;
+                 case TelephonyManager.CALL_STATE_IDLE:
+                     // Phone is idle before and after phone call.
+                     // If running on version older than 19 (KitKat),
+                     // restart activity when phone call ends.
+                     Log.i("onCallStateChanged", "CALL_STATE_IDLE");
+                     break;
+                 default:
+                     Log.i("onCallStateChanged", "default");
+                     break;
+             }
+         }
+    }
+
+    private class CallStateListener extends TelephonyCallback implements TelephonyCallback.CallStateListener {
+
+          @Override
+          public void onCallStateChanged(int state) {
+              switch (state) {
+                  case TelephonyManager.CALL_STATE_RINGING:
+                      // Incoming call is ringing (not used for outgoing call).
+                      Log.i("onCallStateChanged", "CALL_STATE_RINGING");
```

Comment:
Are those debug logs useful ? Eg, all `Log.i()` doesn't seem to provide meaningful information

Reply:
Not a big deal, they just tell us where we are. want me to remove them ?

Reply:
@manuquentin

Reply:
Yes because when you make a `adb logcat` you don't want to be flooded by un-meaningful logs